python/cython_build_ext.py

```diff
@@ -59,46 +64,74 @@ class cython_build_ext(_build_ext, object):
         otherwise be retrieved after compilation.
-    cython_exclude : list of str
-        When passing glob patterns as module_list, you can exclude certain
-        module names explicitly by passing them into the exclude option.
     gdb_debug : bool, default=False
         Passes the `gdb_debug` argument to `cythonize()`. Setting up debugging
         for Cython can be difficult. See the debugging docs here
         https://cython.readthedocs.io/en/latest/src/userguide/debugging.html
+    language_level : {"2", "3", "3str"}, default="2"
+        Globally set the Python language level to be used for module
+        compilation. Default is compatibility with Python 2. To enable Python 3
+        source code semantics, set this to 3 (or 3str)
+    linetrace : bool, default=False
+        Write line tracing hooks for Python profilers or coverage reporting
+        into the compiled C code. This also enables profiling. Default is
+        False. Note that the generated module will not actually use line
+        tracing, unless you additionally pass the C macro definition
+        ``CYTHON_TRACE=1`` to the C compiler (e.g. using the setuptools option
+        define_macros). Define ``CYTHON_TRACE_NOGIL=1`` to also include nogil
+        functions and sections.
+    profile : bool, default=False
+        Write hooks for Python profilers into the compiled C code.
     """
     user_options = [
-        ('language-level=', None,
-         'Sets the python language syntax to use "2", "3", "3str".'),
-        ("binding", None,
+        ("annotate=",
+         None,
+         "Passes the `annotate` argument to `cythonize()`. See the Cython "
+         "docs for more info."),
+        ("binding",
+         None,
          "Sets the binding Cython compiler directive. See the Cython docs for "
          "more info."),
-        ("profile", None,
-         "Sets the profile Cython compiler directive. See the Cython docs for "
-         "more info."),
-        ("embedsignature", None,
-         "Sets the `embedsignature` Cython compiler directive. See the Cython "
-         "docs for more info."),
-        ("cython-exclude=", None,
+        ("cython-exclude=",
+         None,
          "Sets the exclude argument for `cythonize()`. See the Cython docs for"
          " more info."),
-        ("gdb-debug=", None,
+        ("embedsignature",
+         None,
+         "Sets the `embedsignature` Cython compiler directive. See the Cython "
+         "docs for more info."),
+        ("gdb-debug=",
+         None,
          "Passes the `gdb_debug` argument to `cythonize()`. See the Cython "
-         "docs for more info.")
+         "docs for more info."),
+        ('language-level=',
+         None,
+         'Sets the python language syntax to use "2", "3", "3str".'),
+        ("linetrace=",
+         None,
+         "Passes the `linetrace` argument to `cythonize()`. See the Cython "
+         "docs for more info."),
+        ("profile",
```

Comment:
Did you add the profile setting for a particular goal or just because it was easy enough to add? I always love a good profile but wonder whether such a low % of our runtime is in cython that it might not be that helpful.

Reply:
From the [docs](https://cython.readthedocs.io/en/latest/src/tutorial/profiling_tutorial.html#enabling-line-tracing), adding linetracing enables profiling as well. The two options are very similarly related so I added support for both.

Also, while it looks like profiling was added in this PR, the option already existed (it's in 0.16). I just reorganized all of the properties to be listed in alphabetical order which is why you are seeing the diff.

Finally, this class `cython_build_ext`, is meant to be a reusable component that other rapids libraries can drop into their build system and immediately enable code coverage. (This class is something the Cython team should have built themselves, but that's a longer discussion)